### DISCUSSION TITLE
can`t change style on view

### DISCUSSION THREAD

**Discussion by capoia** [GrapesJS/grapesjs#5310] [Q&A]
Hello, I have a component called container, and I need it to have a min-height only inside the grapes, I tried to add it inside the component definition inside view{} but it seems to ignore any way I try to do that.
How I'm trying (I've tried to keep only one of the solutions, it doesn't work, example listing everything I've tried):
![CleanShot 2023-08-16 at 16 17 16@2x](https://github.com/GrapesJS/grapesjs/assets/21159928/e81f2f0a-2cde-43cc-b338-fd67c2cf5495)

How it renders the component inside grapes:
![CleanShot 2023-08-16 at 16 18 11@2x](https://github.com/GrapesJS/grapesjs/assets/21159928/a80e840b-c65f-4f91-8b73-12cb0e5f40b8)

**Comment by capoia** [comment]
I managed to make it work that way, but I find it curious not to be able to do it using the properties.
I'll leave the topic open to understand the correct way to do it.

```
onRender({ el }) {
  el.setAttribute('style', 'min-height: 30px;')
},
```

**Reply by nguyenvoanhkhoa1** [reply]
Base on the [document](https://grapesjs.com/docs/modules/Components.html#components-css) the 
```javascript
styles: `
`
``` 
attribute in the default should work, and work as class name. You can try to:
Add style attribute in the "attributes" attribute :).
Or
Add class to component and write style in style in default.
should do the second solution base on docs

**Comment by artf** [comment]
There are no `style`/`styles` properties of the ComponentView. 
Your solution is ok but the attribute might be removed/updated on some attribute change triggered from the model.
Probably the more reliable way would be to extend `view.onAttrUpdate`:

```js
onAttrUpdate() {
 this.el.style.minHeight = '...';
}
```
